### PR TITLE
Added support of XMM&GP registers for Windows x86 custom callconv

### PR DESCRIPTION
**Pull request by qubka**
In addition added support of some GP registers for making calls with custom call conventions on Windows x86 operating systems. In addition, this code require addional changes in sourcehook.h in the metamod core.

### CODE REVIEW

File: extensions/sdktools/vcaller.cpp [hunk 1]
```diff
@@ -243,7 +243,16 @@ static cell_t PrepSDKCall_AddParameter(IPluginContext *pContext, const cell_t *p
 	DecodePassMethod(info->vtype, method, info->type, info->flags);
 	info->decflags = params[3] | VDECODE_FLAG_BYREF;
 	info->encflags = params[4];
-
+#ifdef PLATFORM_WINDOWS
+	// Validate that the parameter exist, otherwise reset to default
+	if(params[0] == 5)
```
Comment (KyleSanderson) [inline extensions/sdktools/vcaller.cpp]: >=

File: public/extensions/IBinTools.h [hunk 2]
```diff
@@ -53,7 +53,7 @@ namespace SourceMod
 	enum CallConvention
 	{
 		CallConv_ThisCall,		/**< This call (object pointer required) */
-		CallConv_Cdecl,			/**< Standard C call */
```
Comment (KyleSanderson) [inline public/extensions/IBinTools.h]: Leave the ','

**Comment by qubka** [comment]
> I'm unclear on why a new PR was opened.
when I tried to rebase and merge other commits, its close my prev PR, therefore, I opened a new one, sorry about that

File: extensions/sdktools/vcaller.cpp (hunk 1 above)
Comment (qubka) [reply extensions/sdktools/vcaller.cpp]: Sorry but I didn't understand what's wrong in that place, can you explain more detaily?

**Comment by KyleSanderson** [comment]
@qubka can you rebase / 2020ize this one please and ping me when it's ready. I think we need david to help us out a bit here but I'll absolutely take a look when it's ready again.

**Comment by qubka** [comment]
> @qubka can you rebase / 2020ize this one please and ping me when it's ready. I think we need david to help us out a bit here but I'll absolutely take a look when it's ready again.
I don't need this feature anymore, so if you need it, it up to you. GL

**Pull request closed without merging**